src/middlewares/authentication.py

```diff
@@ -0,0 +1,77 @@
+import jwt
+from django.conf import settings
+from rest_framework import exceptions, status
+from django.contrib.auth import get_user_model
+from django.http import HttpResponse
+from rest_framework.authentication import get_authorization_header, BaseAuthentication
+from django.middleware.csrf import CsrfViewMiddleware
+from django.utils.translation import gettext_lazy as _
+
+class CSRFCheck(CsrfViewMiddleware):
+    def _reject(self, request, reason):
+        return reason
+
+
+class AuthenticationJWT(BaseAuthentication):
+    model = get_user_model()
+
+    def authenticate(self, request):
+        auth = get_authorization_header(request).split()
+        if not auth or auth[0].lower() != b'token':
+            return None
+
+        if len(auth) == 1:
+            msg = 'Invalid token header. No credentials provided.'
+            raise exceptions.AuthenticationFailed(msg)
+        elif len(auth) > 2:
```

Comment:
what is the format of correct auth? I don't know when u check len(auth) must = 2?

Reply:
please use regex to verify token's format

Reply:
i have realized the function get_authorization_header in django restframwork make sure the token from header is valid